### PR TITLE
Fix #572: Don't forcefully append .0 decimal when calling toJSON

### LINKED ISSUE #572: Regression in toJSON
In the newer builds after our update to MTASA server 1.5.6, toJSON no longer works as expected.

Large integers passed to toJSON will now return a float, which differs from the expected and previous behavior.

```lua
toJSON(1234567890000) -- four zeroes
```

Outputs incorrectly by converting my integer to a float `[ 1234567890000.0 ]`

I'm unfamiliar with the codebase so bear with me.
The check here: https://github.com/multitheftauto/mtasa-blue/blame/40118f7875516304ff585c4c9ef360a7e887ca41/Client/mods/deathmatch/logic/lua/CLuaArgument.cpp#L912

Maybe fails?

https://github.com/multitheftauto/mtasa-blue/blame/40118f7875516304ff585c4c9ef360a7e887ca41/Shared/sdk/SharedUtil.Math.h#L47

Perhaps the value is too large? But this worked with 1.5.5, also it needs to work for larger ints because of the way we use it for milliseconds since epoch.

### PR DESCRIPTION
**Pull request by patrikjuvonen**
Would be great to know if someone sees any edge problem with this... feels like an awfully ugly fix, but we've seen those ones before haven't we.

The following behaves like before now:

```lua
local strData = "[[[1,1.0,1.1,1.123],[12,12.0,12.1,12.123],[123,123.0,123.1,123.123],[1234,1234.0,1234.1,1234.123],[12345,12345.0,12345.1,12345.123],[123456,123456.0,123456.1,123456.123],[1234567,1234567.0,1234567.1,1234567.123],[12345678,12345678.0,12345678.1,12345678.123],[123456789,123456789.0,123456789.1,123456789.123],[1234567890,1234567890.0,1234567890.1,1234567890.123],[12345678901,12345678901.0,12345678901.1,12345678901.123],[123456789012,123456789012.0,123456789012.1,123456789012.123],[1234567890123,1234567890123.0,1234567890123.1,1234567890123.123],[12345678901234,12345678901234.0,12345678901234.1,12345678901234.123],[123456789012345,123456789012345.0,123456789012345.1,123456789012345.123],[1234567890123456,1234567890123456.0,1234567890123456.1,1234567890123456.123],[12345678901234567,12345678901234567.0,12345678901234567.1,12345678901234567.123],[123456789012345678,123456789012345678.0,123456789012345678.1,123456789012345678.123],[1234567890123456789,1234567890123456789.0,1234567890123456789.1,1234567890123456789.123],[12345678901234567890,12345678901234567890.0,12345678901234567890.1,12345678901234567890.123]]]"
local data = {{1,1.0,1.1,1.123},{12,12.0,12.1,12.123},{123,123.0,123.1,123.123},{1234,1234.0,1234.1,1234.123},{12345,12345.0,12345.1,12345.123},{123456,123456.0,123456.1,123456.123},{1234567,1234567.0,1234567.1,1234567.123},{12345678,12345678.0,12345678.1,12345678.123},{123456789,123456789.0,123456789.1,123456789.123},{1234567890,1234567890.0,1234567890.1,1234567890.123},{12345678901,12345678901.0,12345678901.1,12345678901.123},{123456789012,123456789012.0,123456789012.1,123456789012.123},{1234567890123,1234567890123.0,1234567890123.1,1234567890123.123},{12345678901234,12345678901234.0,12345678901234.1,12345678901234.123},{123456789012345,123456789012345.0,123456789012345.1,123456789012345.123},{1234567890123456,1234567890123456.0,1234567890123456.1,1234567890123456.123},{12345678901234567,12345678901234567.0,12345678901234567.1,12345678901234567.123},{123456789012345678,123456789012345678.0,123456789012345678.1,123456789012345678.123},{1234567890123456789,1234567890123456789.0,1234567890123456789.1,1234567890123456789.123},{12345678901234567890,12345678901234567890.0,12345678901234567890.1,12345678901234567890.123}}

print(inspect(toJSON(data)))
print(inspect(fromJSON(toJSON(data))))

print(inspect(fromJSON(strData)))

print(inspect(toJSON(1234567890000)))
print(inspect(fromJSON(toJSON(1234567890000))))
```

Before json-c update:
```
[2019-01-11 11:49:59] INFO: "[ [ [ 1, 1, 1.1, 1.123 ], [ 12, 12, 12.1, 12.123 ], [ 123, 123, 123.1, 123.123 ], [ 1234, 1234, 1234.1, 1234.123 ], [ 12345, 12345, 12345.1, 12345.123 ], [ 123456, 123456, 123456.1, 123456.123 ], [ 1234567, 1234567, 1234567.1, 1234567.123 ], [ 12345678, 12345678, 12345678.1, 12345678.123 ], [ 123456789, 123456789, 123456789, 123456789 ], [ 1234567890, 1234567890, 1234567890, 1234567890 ], [ 12345678901, 12345678901, 12345678901.1, 12345678901.123 ], [ 123456789012, 123456789012, 123456789012.1, 123456789012.123 ], [ 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.123 ], [ 12345678901234, 12345678901234, 12345678901234.1, 12345678901234.12 ], [ 123456789012345, 123456789012345, 123456789012345.1, 123456789012345.1 ], [ 1234567890123456, 1234567890123456, 1234567890123456, 1234567890123456 ], [ 1.234567890123457e+16, 1.234567890123457e+16, 1.234567890123457e+16, 1.234567890123457e+16 ], [ 1.234567890123457e+17, 1.234567890123457e+17, 1.234567890123457e+17, 1.234567890123457e+17 ], [ 1.234567890123457e+18, 1.234567890123457e+18, 1.234567890123457e+18, 1.234567890123457e+18 ], [ 1.234567890123457e+19, 1.234567890123457e+19, 1.234567890123457e+19, 1.234567890123457e+19 ] ] ]"
[2019-01-11 11:49:59] INFO: { { 1, 1, 1.1, 1.123 }, { 12, 12, 12.1, 12.123 }, { 123, 123, 123.1, 123.123 }, { 1234, 1234, 1234.1, 1234.123 }, { 12345, 12345, 12345.1, 12345.123 }, { 123456, 123456, 123456.1, 123456.123 }, { 1234567, 1234567, 1234567.1, 1234567.123 }, { 12345678, 12345678, 12345678.1, 12345678.123 }, { 123456789, 123456789, 123456789, 123456789 }, { 1234567890, 1234567890, 1234567890, 1234567890 }, { 12345678901, 12345678901, 12345678901.1, 12345678901.123 }, { 123456789012, 123456789012, 123456789012.1, 123456789012.12 }, { 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.1 }, { 12345678901234, 12345678901234, 12345678901234, 12345678901234 }, { 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14 }, { 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15 }, { 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16 }, { 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17 }, { 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18 }, { 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19 } }
[2019-01-11 11:49:59] INFO: { { 1, 1, 1.1, 1.123 }, { 12, 12, 12.1, 12.123 }, { 123, 123, 123.1, 123.123 }, { 1234, 1234, 1234.1, 1234.123 }, { 12345, 12345, 12345.1, 12345.123 }, { 123456, 123456, 123456.1, 123456.123 }, { 1234567, 1234567, 1234567.1, 1234567.123 }, { 12345678, 12345678, 12345678.1, 12345678.123 }, { 123456789, 123456789, 123456789.1, 123456789.123 }, { 1234567890, 1234567890, 1234567890.1, 1234567890.123 }, { 12345678901, 12345678901, 12345678901.1, 12345678901.123 }, { 123456789012, 123456789012, 123456789012.1, 123456789012.12 }, { 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.1 }, { 12345678901234, 12345678901234, 12345678901234, 12345678901234 }, { 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14 }, { 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15 }, { 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16 }, { 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17 }, { 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18 }, { 9.2233720368548e+18, 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19 } }
[2019-01-11 11:49:59] INFO: "[ 1234567890000 ]"
[2019-01-11 11:49:59] INFO: 1234567890000
```

After json-c update:
```
[2019-01-11 11:53:48] INFO: "[ [ [ 1, 1, 1.1, 1.123 ], [ 12, 12, 12.1, 12.123 ], [ 123, 123, 123.1, 123.123 ], [ 1234, 1234, 1234.1, 1234.123 ], [ 12345, 12345, 12345.1, 12345.123 ], [ 123456, 123456, 123456.1, 123456.123 ], [ 1234567, 1234567, 1234567.1, 1234567.123 ], [ 12345678, 12345678, 12345678.1, 12345678.123 ], [ 123456789, 123456789, 123456789, 123456789 ], [ 1234567890, 1234567890, 1234567890, 1234567890 ], [ 12345678901.0, 12345678901.0, 12345678901.1, 12345678901.123 ], [ 123456789012.0, 123456789012.0, 123456789012.1, 123456789012.123 ], [ 1234567890123.0, 1234567890123.0, 1234567890123.1, 1234567890123.123 ], [ 12345678901234.0, 12345678901234.0, 12345678901234.1, 12345678901234.12 ], [ 123456789012345.0, 123456789012345.0, 123456789012345.1, 123456789012345.1 ], [ 1234567890123456.0, 1234567890123456.0, 1234567890123456.0, 1234567890123456.0 ], [ 1.234567890123457e+16, 1.234567890123457e+16, 1.234567890123457e+16, 1.234567890123457e+16 ], [ 1.234567890123457e+17, 1.234567890123457e+17, 1.234567890123457e+17, 1.234567890123457e+17 ], [ 1.234567890123457e+18, 1.234567890123457e+18, 1.234567890123457e+18, 1.234567890123457e+18 ], [ 1.234567890123457e+19, 1.234567890123457e+19, 1.234567890123457e+19, 1.234567890123457e+19 ] ] ]"
[2019-01-11 11:53:48] INFO: { { 1, 1, 1.1, 1.123 }, { 12, 12, 12.1, 12.123 }, { 123, 123, 123.1, 123.123 }, { 1234, 1234, 1234.1, 1234.123 }, { 12345, 12345, 12345.1, 12345.123 }, { 123456, 123456, 123456.1, 123456.123 }, { 1234567, 1234567, 1234567.1, 1234567.123 }, { 12345678, 12345678, 12345678.1, 12345678.123 }, { 123456789, 123456789, 123456789, 123456789 }, { 1234567890, 1234567890, 1234567890, 1234567890 }, { 12345678901, 12345678901, 12345678901.1, 12345678901.123 }, { 123456789012, 123456789012, 123456789012.1, 123456789012.12 }, { 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.1 }, { 12345678901234, 12345678901234, 12345678901234, 12345678901234 }, { 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14 }, { 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15 }, { 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16 }, { 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17 }, { 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18 }, { 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19 } }
[2019-01-11 11:53:48] INFO: { { 1, 1, 1.1, 1.123 }, { 12, 12, 12.1, 12.123 }, { 123, 123, 123.1, 123.123 }, { 1234, 1234, 1234.1, 1234.123 }, { 12345, 12345, 12345.1, 12345.123 }, { 123456, 123456, 123456.1, 123456.123 }, { 1234567, 1234567, 1234567.1, 1234567.123 }, { 12345678, 12345678, 12345678.1, 12345678.123 }, { 123456789, 123456789, 123456789.1, 123456789.123 }, { 1234567890, 1234567890, 1234567890.1, 1234567890.123 }, { 12345678901, 12345678901, 12345678901.1, 12345678901.123 }, { 123456789012, 123456789012, 123456789012.1, 123456789012.12 }, { 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.1 }, { 12345678901234, 12345678901234, 12345678901234, 12345678901234 }, { 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14 }, { 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15 }, { 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16 }, { 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17 }, { 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18 }, { 9.2233720368548e+18, 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19 } }
[2019-01-11 11:53:48] INFO: "[ 1234567890000.0 ]"
[2019-01-11 11:53:48] INFO: 1234567890000
```

After test fix:
```
[2019-01-11 11:51:07] INFO: "[ [ [ 1, 1, 1.1, 1.123 ], [ 12, 12, 12.1, 12.123 ], [ 123, 123, 123.1, 123.123 ], [ 1234, 1234, 1234.1, 1234.123 ], [ 12345, 12345, 12345.1, 12345.123 ], [ 123456, 123456, 123456.1, 123456.123 ], [ 1234567, 1234567, 1234567.1, 1234567.123 ], [ 12345678, 12345678, 12345678.1, 12345678.123 ], [ 123456789, 123456789, 123456789, 123456789 ], [ 1234567890, 1234567890, 1234567890, 1234567890 ], [ 12345678901, 12345678901, 12345678901.1, 12345678901.123 ], [ 123456789012, 123456789012, 123456789012.1, 123456789012.123 ], [ 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.123 ], [ 12345678901234, 12345678901234, 12345678901234.1, 12345678901234.12 ], [ 123456789012345, 123456789012345, 123456789012345.1, 123456789012345.1 ], [ 1234567890123456, 1234567890123456, 1234567890123456, 1234567890123456 ], [ 1.234567890123457e+16, 1.234567890123457e+16, 1.234567890123457e+16, 1.234567890123457e+16 ], [ 1.234567890123457e+17, 1.234567890123457e+17, 1.234567890123457e+17, 1.234567890123457e+17 ], [ 1.234567890123457e+18, 1.234567890123457e+18, 1.234567890123457e+18, 1.234567890123457e+18 ], [ 1.234567890123457e+19, 1.234567890123457e+19, 1.234567890123457e+19, 1.234567890123457e+19 ] ] ]"
[2019-01-11 11:51:07] INFO: { { 1, 1, 1.1, 1.123 }, { 12, 12, 12.1, 12.123 }, { 123, 123, 123.1, 123.123 }, { 1234, 1234, 1234.1, 1234.123 }, { 12345, 12345, 12345.1, 12345.123 }, { 123456, 123456, 123456.1, 123456.123 }, { 1234567, 1234567, 1234567.1, 1234567.123 }, { 12345678, 12345678, 12345678.1, 12345678.123 }, { 123456789, 123456789, 123456789, 123456789 }, { 1234567890, 1234567890, 1234567890, 1234567890 }, { 12345678901, 12345678901, 12345678901.1, 12345678901.123 }, { 123456789012, 123456789012, 123456789012.1, 123456789012.12 }, { 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.1 }, { 12345678901234, 12345678901234, 12345678901234, 12345678901234 }, { 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14 }, { 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15 }, { 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16 }, { 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17 }, { 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18 }, { 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19 } }
[2019-01-11 11:51:07] INFO: { { 1, 1, 1.1, 1.123 }, { 12, 12, 12.1, 12.123 }, { 123, 123, 123.1, 123.123 }, { 1234, 1234, 1234.1, 1234.123 }, { 12345, 12345, 12345.1, 12345.123 }, { 123456, 123456, 123456.1, 123456.123 }, { 1234567, 1234567, 1234567.1, 1234567.123 }, { 12345678, 12345678, 12345678.1, 12345678.123 }, { 123456789, 123456789, 123456789.1, 123456789.123 }, { 1234567890, 1234567890, 1234567890.1, 1234567890.123 }, { 12345678901, 12345678901, 12345678901.1, 12345678901.123 }, { 123456789012, 123456789012, 123456789012.1, 123456789012.12 }, { 1234567890123, 1234567890123, 1234567890123.1, 1234567890123.1 }, { 12345678901234, 12345678901234, 12345678901234, 12345678901234 }, { 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14, 1.2345678901235e+14 }, { 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15, 1.2345678901235e+15 }, { 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16, 1.2345678901235e+16 }, { 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17, 1.2345678901235e+17 }, { 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18, 1.2345678901235e+18 }, { 9.2233720368548e+18, 1.2345678901235e+19, 1.2345678901235e+19, 1.2345678901235e+19 } }
[2019-01-11 11:51:07] INFO: "[ 1234567890000 ]"
[2019-01-11 11:51:07] INFO: 1234567890000
```

### CODE REVIEW

**Comment by qaisjp** [comment]
In #572 you said:

> According to @ccw808 it indeed has to do with the initially referenced ShouldUseInt math method which initially wasn't made to support doubles, but should be adjusted to support them properly. So this is a separate issue, but good to realize now while messing with the stuff.

Is that no longer a solution?

**Comment by patrikjuvonen** [comment]
> In #572 you said:
> 
> > According to @ccw808 it indeed has to do with the initially referenced ShouldUseInt math method which initially wasn't made to support doubles, but should be adjusted to support them properly. So this is a separate issue, but good to realize now while messing with the stuff.
> 
> Is that no longer a solution?

That's a different issue:
```
{
    123456789,    -- becomes 123456789
    123456789.0,  -- becomes 123456789
    123456789.1,  -- becomes 123456789
    123456789.123 -- becomes 123456789
},
{
    1234567890,    -- becomes 1234567890
    1234567890.0,  -- becomes 1234567890
    1234567890.1,  -- becomes 1234567890
    1234567890.123 -- becomes 1234567890
}
```

Not related to the regression, but needs to be fixed (separate issue though). This was an issue also before the json-c update, which I only found when testing the original .0 decimal issue.

**Comment by qaisjp** [comment]
> That's a different issue:
> `...`
> Not related to the regression, but needs to be fixed (separate issue though). This was an issue also before the json-c update, which I only found when testing the original .0 decimal issue.

Ah, ok. :+1: 

I can't really see the difference between _before update_ and _after update_, but since the difference is only a trailing `.0` (i.e `"[ 1234567890000.0 ]"` vs `"[ 1234567890000 ]"`) **imo** I don't think this really needs to be fixed. My reasoning for this is:

> The JSON **data** is the same, it doesn't matter that the string representation is different. As long as data in machine form (after parsing) is identical, it should be fine.

**However, since we're hoping to move away from json-c, we might as well apply your fix just to keep consistency until that big update. That way we can deal with any changes all in one go.**

What do you think?

tl;dr looks fine to merge. what do you think?

**Comment by patrikjuvonen** [comment]
I agree that there is no need to fix it, and in my opinion the issue is trivial at best and could even be closed.

I'll leave the decision of merging this PR and closing the original issue to your discretion. In my opinion it is ok if we leave it as it is in the current release build -- but if it affects some few developers still today, we might as well merge it in and worry about these issues once we swap out the json-c library.

**Review by qaisjp — APPROVED** [review]

**Review by ccw808 — APPROVED** [review]
